skfda/tests/test_fdata_cov.py

```diff
@@ -0,0 +1,186 @@
+"""Test the covariance method of FData."""
+from typing import Any, Callable, Tuple
+
+import numpy as np
+import pytest
+
+from skfda.datasets import make_gaussian_process
+from skfda.misc.covariances import CovarianceLike, Gaussian
+from skfda.representation import FDataBasis
+from skfda.representation.basis import Basis, BSplineBasis, FourierBasis
+
+############
+# FIXTURES
+############
+
+
+@pytest.fixture(
+    params=[
+        FourierBasis,
+        BSplineBasis,
+    ],
+    ids=[
+        "FourierBasis",
+        "BSplineBasis",
+    ],
+)
+def basis_type(
+    request: Any,
+) -> Any:
+    """Fixture for classes to test."""
+    return request.param
+
+
+@pytest.fixture(
+    params=[
+        5,
+        30,
+    ],
+)
+def n_basis(
+    request: Any,
+) -> Any:
+    """Generate a basis."""
+    return request.param
+
+
+@pytest.fixture(
+    params=[
+        100,
+    ],
+)
+def n_samples(
+    request: Any,
+) -> Any:
+    """Generate number of sample."""
+    return request.param
+
+
+@pytest.fixture(
+    params=[
+        100,
+    ],
+)
+def n_features(
+    request: Any,
+) -> Any:
+    """Generate number of features (points of evaluation)."""
+    return request.param
+
+
+@pytest.fixture(
+    params=[
+        (0.0, 1.0),
+        (-2.0, 2.0),
+    ],
+)
+def interval(
+    request: Any,
+) -> Any:
+    """Generate an interval."""
+    return request.param
+
+
+@pytest.fixture(
+    params=[
+        0.0,
+    ],
+)
+def mean(
+    request: Any,
+) -> Any:
+    """Generate a mean."""
+    return request.param
+
+
+@pytest.fixture(
+    params=[
+        0.1,
+    ],
+)
+def noise(
```

Comment:
Why do you test with noise? With noiseless data it is easier to use tests where you know the true answer.